contents/docker/Docker.md

```diff
@@ -0,0 +1,165 @@
+
+#CMD - Identifies the command that should be used by default when running the image as a container.
+CMD ["cat", "/tmp/hello_docker.txt"]
+```
+
+As you can see from above, a DockerFile is a series of instructions that is used to build the container. You start with a base image (ubuntu for the example above) and then add more “layers” to the image, with layers representing a portion of the images file system that either adds to or replaces the layer above it. 
+
+For the example used, the layers are simple bash commands that help print “Hello Docker”. A DockerFile is also able to do much more than this! It can install specific dependencies, it can run a couple of servers, set up configuration files etc. There are plenty of guides available out there that covers the fundamentals behind getting started with this (refer to next section for some links).
+
+As a result, Docker is easy to get started with. Once you have configured a DockerFile, you can reuse it everywhere else to run your container! The process for creating a DockerFile is also made easier due to the plethora of resources available and the Docker Hub.
+
+#### Benefit: Docker is modular and scalable
+
+As mentioned earlier, Docker makes it much easier to deploy an application that is using a microservices based architecture. For example, you may have Postgres database running in one container, Redis server in another and a Python Flask application in another. Docker makes it much easier to group these containers together and scale/update individual components easily in the future.
+
+In order to provide a little more clarity to this, let us consider a simple blog application that is running using Nginx, WordPress and MariaDB. We can organise this as follows:
+
+<img src="Docker_MicroServices.jpg" alt="Docker Microservice Architecture" width="100%">
+
+<sub>_Figure 6. Docker Architecture_ [(source)](https://rubygarage.org/blog/advantages-of-using-docker-for-microservices)</sub>
+
+Each of the above services is encapsulated in a container using Docker. Docker provides an added functionality called `docker-compose` that allows you to run all the containers at once. `docker-compose` also has added advantages:
+- Preserve volume data when containers are created
+- Only recreate containers that have changed
+- Variables and moving a composition between environments
+
+The set-up to use `docker-compose` involves the creation of a YAML file. Detailed information on this is available [here](https://docs.docker.com/compose/overview/).
+
+### How to use Docker?
+
+At this point, you would have realised the advantages that Docker is able to give you and might be considering to switch over to Docker for your own projects. 
+
+Here are some resources that would aid you in this process:
+- https://docs.docker.com/get-started/
+- https://scotch.io/tutorials/getting-started-with-docker
+- https://medium.freecodecamp.org/a-beginner-friendly-introduction-to-containers-vms-and-docker-79a9e3e119b 
```

Comment:
Can be a bit more detailed. e.g.,
> * [Docker's own documentation on how to get started](https://docs.docker.com/get-started/) - {Give your own assessment of this resource}

Also, try to make this a _learning path_ (i.e., in your opinion, this is the order in which the reader should go) rather than just a bunch of links.

Reply:
Yes! Made it into a learning path as suggested